Xamarin.PropertyEditing/ViewModels/ImageBrushViewModel.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Xamarin.PropertyEditing.Drawing;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Xamarin.PropertyEditing.ViewModels
+{
+	internal class ImageBrushViewModel : NotifyingObject
+	{
+		public ImageBrushViewModel (BrushPropertyViewModel parent)
+		{
+			Parent = parent ?? throw new ArgumentNullException (nameof (parent));
+			parent.PropertyChanged += Parent_PropertyChanged;
+		}
+
+		public CommonImageBrush PreviousImageBrush { get; set; }
+
+		public CommonTileMode TileMode {
+			get => Parent.Value is CommonImageBrush imageBrush ? imageBrush.TileMode : CommonTileMode.None;
+			set {
+				if (TileMode != value) {
+					Parent.Value = (Parent.Value as CommonImageBrush ?? new CommonImageBrush()).CopyWith (tileMode: value);
+					OnPropertyChanged ();
+				}
+			}
+		}
+
+		public IEnumerable<KeyValuePair<CommonTileMode, string>> TileModeValues
```

Comment:
We need to consider that the target platforms may not use the same terms exposed in our common types and as such we need to provide some kind of translation here.